packages/wonder-blocks-core/components/unique-id-provider.md

```diff
@@ -0,0 +1,101 @@
+The `UniqueIDProvider` component is how Wonder Blocks components obtain unique identifiers. This component ensures that server-side rendering and initial client rendering match while allowing the provision of unique identifiers for the client. It does this by not guaranteeing unique identifiers for the initial render, but instead allowing for a placeholder to be rendered in place of the real content, or for non-unique values to be provided as identifiers.
+
+In all but the first render, the children are rendered with the same `IIdentifierFactory` instance, ensuring that the same calls will return the same identifiers.
+
+### mockOnFirstRender absent or false
+
+When no `mockOnFirstRender` is `false` (the default), the `children` prop is only called after the initial render. Each call provides the same identifier factory, meaning the same identifier gets returned. Try it below.
+
+```jsx
+const {Body, HeadingSmall} = require("@khanacademy/wonder-blocks-typography");
+const {Spring, Strut} = require("@khanacademy/wonder-blocks-layout");
+const Button = require("@khanacademy/wonder-blocks-button").default;
+
+let providerRef = null;
+
+const renders = [];
+const provider = (
+    <UniqueIDProvider ref={ref => providerRef = ref}>
+        {ids => {
+            renders.push(ids.id("my-unique-id"));
+            return (
+                <View>
+                    {renders.map((id,i) => (
+                        <Body key={i}>Render {i}: {id}</Body>
+                    ))}
+                </View>
+            );
+        }}
+    </UniqueIDProvider>
+);
+
+const onClick = () => {
+    if (providerRef) {
+        providerRef.forceUpdate();
+    }
+};
+
+<View>
+    <View style={{flexDirection: "row"}}><Button onClick={onClick}>Click Me to Rerender</Button><Spring /></View>
+    <Strut size={16} />
+    <HeadingSmall>The UniqueIDProvider:</HeadingSmall>
+    {provider}
+</View>
+```
+
+### mockOnFirstRender is true
+
+When specifying `mockOnFirstRender` to be `true`, the first render will use a mock identifier factory that doesn't guarantee identifier uniqueness.
```

Comment:
`mockOnFirstRender` doesn't explain what's being mocked, maybe `nonUniqueIdOnFirstRender` or `nonUniqueIdOnMount`.

Reply:
Can you provide a real world example of when people would want to use this prop?

Reply:
See ADR #60. I implemented this to that specification, hence the name of the prop. It also explains why the prop is there. I will carry across some of that info.

I did originally use a different prop name but realised we made a decision when ADR60 was approved, so it would behoove me to follow the decision.

Reply:
I think `mockOnFirstRender` makes sense for `KAProvider` b/c it's providing a mock version of the `KA` object.  It makes less sense for `UniqueIDProvider`, but that discussion should probably happen in the ADR.  ADR's aren't immutable are they?

Reply:
I think once they are accepted, we should see them as immutable and have additional ADR's that change them. Otherwise, I think things get complicated.

However, I think it makes sense to have a way to control that first render. I wonder if, instead of not having it or providing a dummy factory, we should provide `null` for the factory instead. That would be my preference.